### PR TITLE
submit lmod cache job from EB end_hook

### PR DESCRIPTION
**Pull request by smoors**
fixes [AB#21962](https://dev.azure.com/VUB-ICT/3e951b95-f932-4cd3-9681-259422e2c681/_workitems/edit/21962)

### CODE REVIEW

File: src/build_tools/hooks_hydra.py [hunk 1]
```diff
@@ -368,3 +371,19 @@ def pre_module_hook(self, *args, **kwargs):  # pylint: disable=unused-argument
...
+
+def end_hook():
+    """Hook to run shortly before EasyBuild completes"""
+
+    logger = fancylogger.getLogger()
+    fancylogger.logToScreen(True, stdout=True)
+    fancylogger.setLogLevelInfo()
+
+    # submit Lmod cache job
+    if os.getenv('BUILD_TOOLS_RUN_LMOD_CACHE'):
```
Comment (wpoely86) [inline src/build_tools/hooks_hydra.py:384]: where does this var come from?

What happens if you run this outside a job?

Comment (smoors) [reply src/build_tools/hooks_hydra.py:384]: it's set in the .bashrc of vsc10001.
i wanted to provide some flexibility to turn it off if needed

File: bin/submit_build.py [hunk 2]
```diff
@@ -100,7 +100,6 @@ def main():
         "dry-run": ("Do not fetch/install, set debug log level", None, "store_true", False, 'D'),
         "skip-fetch": ("Do not fetch the sources, fail if they are missing", None, "store_true", False, 'n'),
         "bwrap": ("Reinstall via new namespace with bwrap", None, "store_true", False, 'b'),
-        "skip-lmod-cache": ("Do not run Lmod cache after installation", None, "store_true", False, 's'),
```
Comment (wpoely86) [inline bin/submit_build.py:103]: you can keep this and control it with the environment variable? Only if it's useful of course.

File: src/build_tools/hooks_hydra.py (hunk 1 above)
Comment (wpoely86) [reply src/build_tools/hooks_hydra.py:384]: Why not let it be set by the `submit build` script?

File: src/build_tools/hooks_hydra.py [hunk 3]
```diff
@@ -368,3 +371,19 @@ def pre_module_hook(self, *args, **kwargs):  # pylint: disable=unused-argument
...
+        partition = os.getenv('SLURM_JOB_PARTITION')
```
Comment (wpoely86) [inline src/build_tools/hooks_hydra.py:385]: Can you handle the case if it's not on in a slurm job? Skip or error out or something else?

File: src/build_tools/hooks_hydra.py [hunk 4]
```diff
@@ -381,9 +381,11 @@ def end_hook():
     fancylogger.setLogLevelInfo()
 
     # submit Lmod cache job
-    if os.getenv('BUILD_TOOLS_RUN_LMOD_CACHE'):
+    if os.getenv('BUILD_TOOLS_RUN_LMOD_CACHE', '1'):
```
Comment (wpoely86) [inline src/build_tools/hooks_hydra.py:384]: are you sure this works as intended? Doesn't `export BUILD_TOOLS_RUN_LMOD_CACHE=` return true here?

File: bin/submit_build.py [hunk 5]
```diff
@@ -72,6 +73,7 @@ def main():
     # Easybuild default paths
     # start using environment from local machine, job scripts get custom paths
     ebconf = {
+        'accept-eula-for': 'Intel-oneAPI,CUDA',
```
Comment (wpoely86) [inline bin/submit_build.py:76]: this seems unrelated/unused here?

File: src/build_tools/hooks_hydra.py (hunk 4 above)
Comment (smoors) [reply src/build_tools/hooks_hydra.py:384]: > are you sure this works as intended? Doesn't `export BUILD_TOOLS_RUN_LMOD_CACHE=` return true here?

no, an empty string is not `None`, but still evaluates to `False`.

but i am not against doing a hard `== 1` if you prefer

File: bin/submit_build.py (hunk 5 above)
Comment (smoors) [reply bin/submit_build.py:76]: indeed unrelated, i wanted to do this for a long time, so i sneaked it in.
we currently set this with an envvar in .bashrc, but better to let the script do it.

File: src/build_tools/hooks_hydra.py (hunk 4 above)
Comment (wpoely86) [reply src/build_tools/hooks_hydra.py:384]: I would be inclined to go for a hard `== 1` to have clear behaviour. Only a 1 is sets it, everything else is `False`.

**Review by wpoely86 — APPROVED** [review]
